PureInvoke/Functions/Invoke-KernelFindFileName.ps1

```diff
@@ -0,0 +1,128 @@
+
+function Invoke-KernelFindFileName
+{
+    <#
+    .SYNOPSIS
+    Calls the Win32 `FindFirstFileNameW` and `FindNextFileNameW` functions to get the hardlinks to a file.
+
+    .DESCRIPTION
+    The `Invoke-KernelFindFileName` function finds all the hardlinks to a file. It calls the Win32 `FindFirstFileNameW`
+    and `FindNextFileNameW` functions to get the paths. It returns the path to each hardlink, which includes the path
+    to the file itself. The paths are returned *without* drive qualifiers at the beginning. Since hardlinks can't cross
+    physical file systems, their drives will be the same as the source path.
+
+    .LINK
+    https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-findfirstfilenamew
+
+    .LINK
+    https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-findnextfilenamew
+
+    .EXAMPLE
+    Invoke-KernelFindFileName -Path 'C:\link.txt'
+
+    Demonstrates how to get the hardlinks to a file by passing its path to the `Invoke-KernelFindFileName` function's
+    `Path` parameter.
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [String] $Path
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+
+    [PureInvoke.ErrorCode] $errCode = [PureInvoke.ErrorCode]::Ok
+
+    # Loop over and collect all hard links as their full paths.
+    [IntPtr]$findHandle = [IntPtr]::Zero
+
+    # Must be MAX_PATH otherwise calls to FindFirstFileName@ and FindNextFileNameW throw exceptions that crash

```

Comment:
Typo: `@` should be `W`

Reply:
Removed comment as it doesn't apply to anything.